vermouth/tests/test_selectors.py

```diff
@@ -0,0 +1,148 @@
+    """
+    assert vermouth.selectors.select_all(node)
+
+
+@pytest.mark.parametrize('node, expected', (
+    ({}, False),  # empty node
+    ({'atomname': 'not BB'}, False),  # wrong atomname
+    ({'resname': 'anything'}, False),  # no atomname
+    ({'atomname': 'BB'}, True),  # only correct atomname
+    ({'atomname': 'BB', 'other': 'something'}, True),  # correct atomname + other attribute
+))
+def test_select_backbone(node, expected):
+    """
+    Test that :func:`vermouth.selectors.select_backbone` only select beads with
+    'BB' as atom name.
+    """
+    assert vermouth.selectors.select_backbone(node) == expected
+
+
+@pytest.mark.parametrize('node, attribute, values, expected', (
+    ({}, 'name', ['something', 'other'], False),
+    ({'name': 'nothing'}, 'name', ['something', 'other'], False),
+    ({'name': 'something'}, 'name', ['something', 'other'], True),
+    ({'name': 'nothing', 'plop': 'other'}, 'name', ['something', 'other'], False),
+    ({'name': 'nothing'}, 'name', [], False),
+))
+def test_select_proto_select_attribute_in(node, attribute, values, expected):
+    """
+    Test that :func:`vermouth.selectors.proto_select_attribute_in` works as
+    expected and can be used with :func:`functools.partial`.
+    """
+    # The use of functools.partial is not necessary. Though this is how
+    # proto_select_attribute_in is used in the example from its docstring so
+    # it should work.
+    select_attribute_in = functools.partial(
+        vermouth.selectors.proto_select_attribute_in,
+        attribute=attribute, values=values,
+    )
+    assert select_attribute_in(node) == expected
+
+
+@pytest.mark.parametrize('node, templates, ignore_keys, expected', (
+    ({}, [], (), False),  # Everything empty
+    ({}, [{'name': 'A'}, {'name': 'B'}], (), False),  # Empty node
+    ({'name': 'A'}, [{'resname': 'A'}, {'resid': 1}], (), False),  # Not matching
+    ({'name': 'A'}, [{'name': 'not A'}, {'name': 'A', 'resname': 'something'}], (), False),
+    ({'name': 'A'}, [{'resid': 1, 'resname': 'something'}, {'name': 'A'}], (), True),
+    ({'name': 'A', 'resname': 'plop'}, [{'name': 'A'}], ('resname', 'other'), True),
+))
+def test_proto_multi_templates(node, templates, ignore_keys, expected):
+    """
+    Test that :func:`vermouth.selectors.proto_multi_templates` works as
+    expected and can be used with :func:`functools.partial`.
+    """
+    multi_templates = functools.partial(
+        vermouth.selectors.proto_multi_templates,
+        templates=templates,
+        ignore_keys=ignore_keys,
+    )
+    assert multi_templates(node) == expected
+
+
+def test_filter_minimal():
+    """
+    Test that :func:`vermouth.selectors.filter_minimal` works as expected.
+    """
+    # Build a molecule that has all even atoms with no positions.
+    molecule = read_pdb(str(PDB_PROTEIN))
+    for atom in list(molecule.nodes.values())[::2]:
+        atom['position'] = None
```

Comment:
del atom['position']?

Reply:
It should be equivalent in the way it is handled.

Reply:
You don't get to use the word "should" when talking about tests